prov/efa/src/efa_tostr.c

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright (c) 2022 Amazon.com, Inc. or its affiliates. All rights reserved.
+ *
+ * This software is available to you under a choice of one of two
+ * licenses.  You may choose to be licensed under the terms of the GNU
+ * General Public License (GPL) Version 2, available from the file
+ * COPYING in the main directory of this source tree, or the
+ * BSD license below:
+ *
+ *     Redistribution and use in source and binary forms, with or
+ *     without modification, are permitted provided that the following
+ *     conditions are met:
+ *
+ *      - Redistributions of source code must retain the above
+ *        copyright notice, this list of conditions and the following
+ *        disclaimer.
+ *
+ *      - Redistributions in binary form must reproduce the above
+ *        copyright notice, this list of conditions and the following
+ *        disclaimer in the documentation and/or other materials
+ *        provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "efa_errno.h"
+
+#define UNKNOWN_ERR_STR "Unknown error"
+
+static const char *efa_io_comp_status_str(int status)
+{
+	static const char *status_str[] = {
+		[FI_EFA_OK]                            = "Success",
+		[FI_EFA_FLUSHED]                       = "Flushed during QP destroy",
+		[FI_EFA_LOCAL_ERROR_QP_INTERNAL_ERROR] = "Internal queue pair error",
+		[FI_EFA_LOCAL_ERROR_INVALID_OP_TYPE]   = "Invalid operation type",
+		[FI_EFA_LOCAL_ERROR_INVALID_AH]        = "Invalid address handle",
+		[FI_EFA_LOCAL_ERROR_INVALID_LKEY]      = "Invalid local key (LKEY)",
+		[FI_EFA_LOCAL_ERROR_BAD_LENGTH]        = "Message too long",
+		[FI_EFA_REMOTE_ERROR_BAD_ADDRESS]      = "Invalid address",
+		[FI_EFA_REMOTE_ERROR_ABORT]            = "Remote connection aborted",
+		[FI_EFA_REMOTE_ERROR_BAD_DEST_QPN]     = "Invalid destination queue pair number (QPN)",
+		[FI_EFA_REMOTE_ERROR_RNR]              = "Destination receiver not ready",
```

Comment:
Destination and receiver are basically the same thing, a bit redundant

Reply:
There's a mixture of using "remote, receiver, and destination", just pick one and stick with it